cpp/tests/experimental/louvain_primitives.cu

```diff
@@ -0,0 +1,179 @@
+#include <experimental/graph.hpp>
+
+#include <utilities/shuffle_comm.cuh>
+
+#include <rmm/device_uvector.hpp>
+
+// TODO:
+//    The following 3 include files are not referenced in my file,
+//    however if they are not included prior to the two pattern
+//    includes below then the compilation fails.  I suspect these
+//    should be included somehow in the patterns includes below
+//
+#include <matrix_partition_device.cuh>
+#include <patterns/edge_op_utils.cuh>
+#include <utilities/host_scalar_comm.cuh>
+// END EXTRA INCLUDES
+
+#include <patterns/copy_v_transform_reduce_key_aggregated_out_nbr.cuh>
+#include <patterns/transform_reduce_by_adj_matrix_row_col_key_e.cuh>
+
+template <typename graph_t>
+void update_by_delta_modularity(raft::handle_t const& handle,
+                                graph_t const& graph_view,
+                                std::vector<typename graph_t::vertex_type> cluster_h,
+                                typename graph_t::weight_type total_edge_weight,
+                                typename graph_t::weight_type resolution)
+{
+  using vertex_t = typename graph_t::vertex_type;
+  using edge_t   = typename graph_t::edge_type;
+  using weight_t = typename graph_t::weight_type;
+
+  vertex_t num_verts = graph_view.get_number_of_vertices();
+
+  rmm::device_uvector<vertex_t> cluster_v(num_verts, handle.get_stream());
+
+  raft::update_device(cluster_v.data(), cluster_h.data(), num_verts, handle.get_stream());
+
+  auto tmp = cugraph::experimental::transform_reduce_by_adj_matrix_col_key_e(
```

Comment:
```
-  auto tmp = cugraph::experimental::transform_reduce_by_adj_matrix_col_key_e(
+  rmm::device_uvector<vertex_t> communities(0, handle.get_stream());
+  rmm::device_uvector<weight_t> community_in_weight_sums(0, handle.get_stream());
+  std::tie(communities, community_in_weight_sums) = cugraph::experimental::transform_reduce_by_adj_matrix_col_key_e(
```

Reply:
I knew something like this would work, thanks for providing the syntax I was too lazy to construct 👍 

I integrated this into the code.